### ISSUE TITLE
The http.Client passed to NewBackends isn't being used

### ISSUE THREAD

**Issue by spastorelli-stripe** [stripe/stripe-go#626]
When running on Google AppEngine, the `http.DefaultClient` can't be used so `urlfetch.Client` should be used instead. 

But when using the below code as suggested [in this section of the doc](https://github.com/stripe/stripe-go#google-appengine):
```
c := appengine.NewContext(r)
httpClient := urlfetch.Client(c)
stripeClient.New("sk_test_xxxxx", stripe.NewBackends(httpClient))
```
the requests fail with the following error:
```
Error: Post https://api.stripe.com/v1/charges: http.DefaultTransport and http.DefaultClient are not available in App Engine. See https://cloud.google.com/appengine/docs/go/urlfetch/ 
```
It looks like the `http.Client` instance passed to  `stripe.NewBackends()` isn't actually used anymore:
https://github.com/stripe/stripe-go/blob/d77c49777296d00b9d49ecc2744772913d00f699/stripe.go#L612-L617

which means that the client passed here:
https://github.com/stripe/stripe-go/blob/d77c49777296d00b9d49ecc2744772913d00f699/stripe.go#L534
is the default client created there:
https://github.com/stripe/stripe-go/blob/d77c49777296d00b9d49ecc2744772913d00f699/stripe.go#L731
I believe this would explain the GAE error in this case.

Using v35.9.0 (prior to this [commit](https://github.com/stripe/stripe-go/pull/611/commits/647da4239899ee0f4e024fc0449cfda4f9249559#diff-ef7561d9a44d35bccb615dd5adb151beL514)) allows the `http.Client` passed to `stripe..NewBackends()` to be used and the requests on GAE to succeed.

**Comment by remi-stripe** [comment]
Assigning to @brandur-stripe  to triage/investigate

**Comment by brandur-stripe** [comment]
Oops, this one is my fault. Thanks for logging @spastorelli-stripe and for the detailed information on reproducing! Added a fix in #627.

**Comment by brandur-stripe** [comment]
Patched in 36.1.1.

**Comment by spastorelli-stripe** [comment]
Thanks @brandur-stripe for fixing this quickly.
